src/platform/intel/cavs/lib/power_down.S

```diff
@@ -150,6 +158,17 @@ _PD_SEND_IPC:
 	l32i temp_reg1, host_base, IPC_DIPCTDA
 	or temp_reg1, temp_reg1, temp_reg2
 	s32i temp_reg1, host_base, IPC_DIPCTDA
+#elif CONFIG_IPC_MAJOR_4
+/* Send IPC reply for SET_DX message */
+	movi temp_reg1, 0
+	s32i temp_reg1, host_base, IPC_DIPCIDD
+
+	movi temp_reg1, set_dx_reply
+	l32i temp_reg1, temp_reg1, 0
```

Comment:
cannot the above two asm instructions be merged into
```
l32i temp_reg1, set_dx_reply, 0
```
? And it should be easy to verify by looking at the response on the host side?

Reply:
@lyakh, let me try. I have used the same flow as the existing asm does for `ipc_flag`. It must know better than me ;)

Reply:
ah, I see now, `host_base` is indeed a register, I thought it would be the same syntax as line 150 / 158 above `l32i temp_reg1, host_base, IPC_DIPCTDA` nm, ignore this then, sorry